packages/dds/merge-tree/src/test/obliterate.partialLength.spec.ts

```diff
@@ -0,0 +1,368 @@
+            client.obliterateRange(
+                0,
+                "hello ".length,
+                refSeq,
+                localClientId,
+                refSeq + 1,
+                false,
+                undefined as any,
+            );
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ], refSeq);
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ]);
+        });
+
+        it("passes for local remove and remote obliterate", () => {
+            client.removeRangeLocal(
+                0,
+                "hello ".length,
+            );
+            client.obliterateRange(
+                0,
+                "hello ".length,
+                refSeq,
+                remoteClientId,
+                refSeq + 1,
+                false,
+                undefined as any,
+            );
+
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ]);
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "world".length },
+            ], 0);
+        });
+
+        it("passes for remote remove and local obliterate", () => {
+            client.removeRangeRemote(
+                0,
+                "hello ".length,
+                refSeq + 1,
+                refSeq,
+                client.getLongClientId(remoteClientId),
+            );
+            client.obliterateRange(
+                0,
+                "hello ".length,
+                refSeq,
+                localClientId,
+                refSeq + 2,
+                false,
+                undefined as any,
+            );
+
+            validatePartialLengths(remoteClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ]);
+            validatePartialLengths(localClientId, client.mergeTree, [
+                { seq: refSeq, len: "hello world".length },
+                { seq: refSeq + 1, len: "hello world".length },
+                { seq: refSeq + 2, len: "world".length },
+            ], 0);
+        });
+
+        it("passes for remote remove and remote obliterate", () => {
+            client.removeRangeRemote(
+                0,
+                "hello ".length,
+                refSeq + 1,
+                refSeq,
+                client.getLongClientId(remoteClientId),
+            );
+            client.obliterateRange(
```

Comment:
this should be done with two different remote clients. It also looks like the `seq` on the ops is the same as written, but i'd expect them to be different.